subprojects/model-core/src/main/java/org/gradle/internal/reflect/annotations/impl/DefaultTypeAnnotationMetadataStore.java

```diff
@@ -83,6 +88,12 @@ public void visitValidationFailures(@Nullable String ownerPath, ParameterValidat
         }
     };
 
+    private static final ImmutableSet<? extends Type> MUTABLE_NON_FINAL_TYPES = ImmutableSet.of(
```

Comment:
Is it okay to identify problematic classes in a static way, or should I extend it to use annotations?

Reply:
For now statically defined types should be okay. However, we should keep this class independent of the actual types, and pass them as parameters instead.

Reply:
Could you please be more specific, where would you like me to define these?

Reply:
In `ExecutionGlobalServices`.

Reply:
Done, thanks.